### PR TITLE
Change `linguist-documentation` example from /* to /**

### PR DESCRIPTION
**Pull request by prototriangle**
Users may find this edit helpful as it is more likely to align with their intentions.

>The rules by which the pattern matches paths are the same as in `.gitignore` files (see [gitignore[5]](https://www.git-scm.com/docs/gitignore), with a few exceptions:
>- negative patterns are forbidden
>- patterns that match a directory do not recursively match paths inside that directory **(so using the trailing-slash `path/` syntax is pointless in an attributes file; use `path/**` instead)**

The above was retrieved from [here](https://www.git-scm.com/docs/gitattributes#main).

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Review by Alhadis — COMMENTED** [review]
Attributes are inherited by files inside subdirectories, which means that `foo/*` is, for all intents and purposes, identical to `foo/**`. This might confuse less experienced users, particularly those unfamiliar with how globbing works. At best, it adds no new information. At worst, it's unclear to novices who've seen `*` used, but never `**`, encouraging [cargo-cult](https://en.wikipedia.org/wiki/Cargo_cult_programming) or faulty recollection the next time they need wildcard file-matching.

**TL;DR:** 👎 , sorry.

**Comment by lildude** [comment]
I agree with @Alhadis here. This isn't really necessary, may lead to confusion and may not align with the developer's real intentions.

> Users may find this edit helpful as it is more likely to align with their intentions.

You can't make that assumption. The user may have intended to only apply to the files in that directory and _not_ recursing. See https://github.com/github/linguist/pull/4889#issuecomment-759386273 for another discussion on this.

As with that PR, ultimately, this is really about understanding the git documentation and the developer's end goals, and not making any assumptions, which are outside the scope of using Linguist.

Closing.
